src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPageDefault/hooks/useRelationships.js

```diff
@@ -0,0 +1,178 @@
+// @flow
+import { useCallback, useEffect, useState } from 'react';
+import {
+    getCachedSingleResourceFromKeyAsync,
+} from '../../../../../MetaDataStoreUtils/MetaDataStoreUtils';
+import { getProgramAndStageFromEvent, getTrackedEntityTypeThrowIfNotFound }
+    from '../../../../../metaData';
+import { userStores } from '../../../../../storageControllers/stores';
+import type {
+    InputRelationship, RelationshipData,
+} from '../../../common/EnrollmentOverviewDomain/useCommonEnrollmentDomainData';
+import { getDisplayFieldsFromAPI, getBaseConfigHeaders, relationshipWidgetTypes } from './constants';
+
+/* eslint-disable complexity */
```

Comment:
hey, I'm inclined to keep the condition as it it [here](https://github.com/dhis2/tracker-capture-app/blob/master/components/relationship/relationship-controller.js#L182-L51)  to make sure that we don't have any side effect.

Reply:
hey @simonadomnisoru, I'm looking at many pages (i.e: `EnrollmentEventEditPage` and `EnrollmentAddEventPage`) they all have complexity > 11, are you aware of this issue? Even though there is no flag eslint-disable but it's really hard to read 
<img width="780" alt="Screenshot 2022-02-15 at 07 58 25" src="https://user-images.githubusercontent.com/89806888/154017794-a855e36a-7726-4615-98de-963dd722cb33.png">

Reply:
Hey, 

Thank you for answering my concern. Yes, I was aware that the pages containers have high complexity. This is another issue that we can refactor when/if we have time and prioritize it. 

Regarding the complexity of the function `getRelationshipAttributes` from this PR, my advice still remains to split it and make it as readable and maintainable as possible. I will let the decision up to the rest of the team (@JoakimSM  and @eirikhaugstulen)
Thanks!

Reply:
Hey, 
After a talk with @jasminenguyennn, I agree will her point of view. 
I would like to brainstorm on how to ensure the same maintainability and readability across all the codebase in one of our next FE tech talks. I think this can be beneficial for both the developers and the PR reviews.  
Thank you!